hazelcast-client/src/main/java/com/hazelcast/client/proxy/txn/TransactionProxy.java

```diff
@@ -70,6 +70,9 @@ public State getState() {
 
     void begin() {
         try {
+            if (client.getConnectionManager().getOwnerConnection() == null) {
+                throw new TransactionException("Owner connection needs to be present to begin a transaction");
```

Comment:
Can you add tests for such transactıon exceptions?

Reply:
I have added a new test for  transactions `ClientTxnOwnerDisconnectedTest`